### PR TITLE
git: Add commit template

### PR DESCRIPTION
**Pull request by mzidek-rh**
In order to use the commit template, run the
following command:
git config commit.template .git-commit-template

### CODE REVIEW

**Comment by mzidek-rh** [comment]
This is the same commit template we use in SSSD. Maybe it will be helpful for FreeIPA too.

**Comment by martbab** [comment]
Could you please change the link in the template to:

https://fedorahosted.org/freeipa/ticket/XXXX

just to avoid copy-paste errors when filling in ticket No.?

**Comment by mzidek-rh** [comment]
Sure, that was a copy paste mistake in itself :) . Patch updated.

File: .git-commit-template [hunk 1]
```diff
@@ -0,0 +1,9 @@
+COMPONENT: Subject
```
Comment (jcholast) [inline .git-commit-template]: We don't upper-case the component name.

File: .git-commit-template [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+Explanation
+
+Resolves:
+https://fedorahosted.org/freeipa/ticket/XXXX
```
Comment (jcholast) [inline .git-commit-template]: We don't put "Resolves:" before ticket URL.

File: .git-commit-template [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+# Try to keep the subject line within 52 chars ----|
+# Also please try to not exceed 72 characters of length for the body --|
```
Comment (jcholast) [inline .git-commit-template]: IMO it would be better if these comments were directly above the subject and body, respectively.

**Comment by mzidek-rh** [comment]
Forgot to add a comment. I updated the patch according to jcholast's comments.

**Comment by mbasti-rh** [comment]
I disagree here with Honza,  I liked those comments more at bottom. Why do we even need that comments? git commit command with vim set as editor is doing that automatically.

**Comment by jcholast** [comment]
Not everyone uses vim. I'd like the comments visually close to the relevant commit message part, otherwise the visual guides in them do not seem very useful to me. I'd be fine with having no comments altogether too.

**Comment by mbasti-rh** [comment]
+1 for not having comments there

**Comment by mzidek-rh** [comment]
Comments deleted.

**Comment by mbasti-rh** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/2df709838905dec3ee2c2eaec47f506336d85a6e

**Pull request closed without merging**